src/MassTransit/Context/LogContext.cs

```diff
@@ -0,0 +1,71 @@
+namespace MassTransit.Context
+{
+    using System;
+    using System.Diagnostics;
+    using Logging;
+    using Microsoft.Extensions.Logging;
+
+
+    public partial class LogContext
+    {
+        public static void ConfigureCurrentLogContext(ILoggerFactory loggerFactory = null, DiagnosticSource source = null)
+        {
+            Current = new BusLogContext(loggerFactory ?? NullLoggerFactory.Instance, source ?? Cached.Default.Value);
```

Comment:
Why we couldn't have this one on the Bus level without static?

Reply:
Because there isn't any need to inject it into every single class in the project, when AsyncLocal works just fine.

Reply:
I'm not a big fan of it. clear :)